### PR TITLE
Fixed issue #18459: When filtering the survey list by a parent survey group, surveys assigned to a child survey group are not listed

### PR DESCRIPTION
**Pull request by gabrieljenik**
Tests added.

### CODE REVIEW

**Comment by tiborpacalat** [comment]
I am pretty sure we fixed this already, but in this branch I am getting 500

Screenshot 2023-07-04 at 14 57 40

**Comment by tiborpacalat** [comment]
Then I zip and upload the plugin and come to this:

Screenshot 2023-07-04 at 15 01 09

**Review by Shnoulle — COMMENTED** [review]
Some random string in group name for us ?

File: tests/unit/models/SurveySearchTest.php [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        $this->assertNotEmpty($data, 'The survey search results were unexpectedly empty');
+        $this->assertCount($surveyCount, $data, 'The number of surveys in the resultset found does not match the number of surveys in the database.');
+        $this->assertEquals($surveyCount, $totalItemCount, 'The number of surveys found does not match the number of surveys in the database.');
+
+        $this->assertThat($sids, $this->contains($data[0]->sid), 'One of the expected surveys was not found.');
+        $this->assertThat($sids, $this->contains($data[1]->sid), 'One of the expected surveys was not found.');
+        $this->assertThat($sids, $this->contains($data[2]->sid), 'One of the expected surveys was not found.');
+    }
+
+    private function createGroups()
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply tests/unit/models/SurveySearchTest.php:190]: This test is already merged on master.
v5 will loose support this month. I would just not follow up on this one.

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**